diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,260 @@
+ * comments in if-else
+ */
+class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdoc-comments-codeblocks-formatting") {
+
+    companion object {
+        private const val MAX_SPACES = 1
+        private const val APPROPRIATE_COMMENT_SPACES = 1
+    }
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        isFixMode = autoCorrect
+        emitWarn = emit
+
+        val configuration = KdocCodeBlocksFormattingConfiguration(
+                configRules.getRuleConfig(COMMENT_WHITE_SPACE)?.configuration ?: mapOf())
+
+
+        when (node.elementType) {
+            CLASS, FUN, PROPERTY -> {
+                checkBlankLineAfterKdoc(node, EOL_COMMENT)
+                checkBlankLineAfterKdoc(node, KDOC)
+                checkBlankLineAfterKdoc(node, BLOCK_COMMENT)
+            }
+            IF -> {
+                handleIfElse(node)
+            }
+            EOL_COMMENT -> {
+                handleEolComments(node, configuration)
+            }
+            BLOCK_COMMENT -> {
+                handleBlockComments(node, configuration)
+            }
+            KDOC -> {
+                handleKdocComments(node)
+            }
+        }
+    }
+
+    private fun checkBlankLineAfterKdoc(node: ASTNode, type: IElementType) {
+        val kdoc = node.getFirstChildWithType(type)
+        val nodeAfterKdoc = kdoc?.treeNext
+        if (nodeAfterKdoc?.elementType == ElementType.WHITE_SPACE && nodeAfterKdoc.numNewLines() > 1) {
+            WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, kdoc.text, nodeAfterKdoc.startOffset) {
+                nodeAfterKdoc.leaveOnlyOneNewLine()
+            }
+        }
+    }
+
+    private fun handleKdocComments(node: ASTNode) {
+        if (node.treeParent.treeParent.elementType == BLOCK) {
+            checkCommentsInCodeBlocks(node.treeParent) // node.treeParent is a node that contains a comment.
+        } else if (node.treeParent.elementType != IF){
+            checkClassComment(node)
+        }
+    }
+
+    private fun handleEolComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        checkSpaceBetweenPropertyAndComment(node, configuration)
+
+        if (node.treeParent.elementType == BLOCK && node.treeNext != null) {
+            checkCommentsInCodeBlocks(node)
+        } else if (node.treeParent.lastChildNode != node && node.treeParent.elementType != IF) {
+            checkClassComment(node)
+        }
+
+        checkWhiteSpaceBeforeComment(node)
+    }
+
+    private fun handleBlockComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
```

Comment:
I think, a lot of logic can be extracted from `handleEolComments` and `handleBlockComments`